### DISCUSSION TITLE
Making an OCR

### DISCUSSION THREAD

**Discussion by TheRealKraytonian** [SixLabors/ImageSharp#2602] [Q&A]
I would like your help can you point me toward something?

**Comment by gfoidl** [comment]
Can you be more specific what you want?

"OCR" is generic and broad, so what kind of help do you except (except typing "OCR" in your favorite search engine and go for it -- I hope you understand what and how I mean that).

**Reply by TheRealKraytonian** [reply]
Awesome thank you for getting back to me I am looking at creating a system that process contracts and official id's(school, work, or government). these contracts are 5 to 10 pages with a mixture of multiple hand written inputs such as notes comments initials and signatures. I want to create an orc that identifies the contracts, forms, and id's inputs. I would also like the id to extract all that info and put it in a database in an organized way. Multiple students can have multiple contracts forms and ids. I would like to stick with in the dotnet ecosystem. I have looked at tools like opencv emgucv and tesseract along with ml.net. I am looking for guidance on how I would do this. does it look like I am on the right track. do you need me to go deeper on anything because I will. So far my tech stack is .netMaui c# api, mss and maria db.

**Reply by TheRealKraytonian** [reply]
I am willing to create any thing from scratch just need the right guidence. i cannot afford to pay for these expensive tools.

**Comment by shapeh** [comment]
Hello @TheRealKraytonian 

I have built something similar in the past for a business in the SADC region. I used Azure AI Document Intelligence (used to be called Azure Form Recognizer). AADI supports handwritten notes and also printed text in various languages: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/language-support-ocr?view=doc-intel-4.0.0&tabs=read-hand%2Clayout-hand%2Cgeneral

Read more about it here: 
https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-read?view=doc-intel-4.0.0

I think this would be the best start if you want quick success with a project like this. Using ImageSharp, OpenCV or similar requires considerable amount of coding to get this going.
